### ISSUE TITLE
Epoch 313 Stuck

### ISSUE THREAD

**Issue by CyberCyclone** [input-output-hk/cardano-graphql#648]
I've had 3 out 4 cardano-graphql nodes stop at the boundary to epoch 313 while running on mainnet. Looking at https://explorer.cardano.org/en it seems to be having the same issues (load balancing between in-sync and out-of-sync nodes). I haven't seen anything in the logs to indicate if this is a [@cardano-db-sync](https://github.com/input-output-hk/cardano-db-sync) issue to not.

**Comment by CyberCyclone** [comment]
After bringing the stack down and back up again I get these errors:

```
cardano-db-sync-extended_1  | [db-sync-node:Warning:277] [2022-01-05 23:58:11.07 UTC] validateEpochRewards: rewards spendable in epoch 313 expected total of 14210014.525259 ADA but got 14214107.1934 ADA
```

```
cardano-db-sync-extended_1  | [db-sync-node:Error:277] [2022-01-05 23:59:05.00 UTC] diffRewardMap:
cardano-db-sync-extended_1  | [db-sync-node:Error:277] [2022-01-05 23:59:05.00 UTC]   e100327891bf2d2affca40b5ecbbdbe1612ff0f3a23b6e808a0291ca27: [(RwdMember,DbLovelace 38496)] /= []
cardano-db-sync-extended_1  | [db-sync-node:Error:277] [2022-01-05 23:59:05.00 UTC]   e1005bc4ef3df0e7da073a32057a96029874ca137fe640dbfb422f85b1: [(RwdMember,DbLovelace 833551)] /= []
cardano-db-sync-extended_1  | [db-sync-node:Error:277] [2022-01-05 23:59:05.00 UTC]   e10086894cab8b4a2081f422dbbe77a078aa2b95a5cbaaadcfbe2baabd: [(RwdMember,DbLovelace 418506)] /= []
cardano-db-sync-extended_1  | [db-sync-node:Error:277] [2022-01-05 23:59:05.00 UTC]   e100a146269ea8af0abfebe9d4f5dbfbd5c7453bc24d695211c6c20e95: [(RwdMember,DbLovelace 500003)] /= []
cardano-db-sync-extended_1  | [db-sync-node:Error:277] [2022-01-05 23:59:05.00 UTC]   e100c065b323f54438e84e8f3bd6f29b7b42dae721abf6cd688f7f5c8f: [(RwdMember,DbLovelace 111059)] /= []
cardano-db-sync-extended_1  | [db-sync-node:Error:277] [2022-01-05 23:59:05.00 UTC]   e1012c9c5cead0663731628b07e096a30b3284b761a8a24a04655d7aed: [(RwdMember,DbLovelace 905599)] /= []
cardano-db-sync-extended_1  | [db-sync-node:Error:277] [2022-01-05 23:59:05.00 UTC]   e1016465762d6ed7ed4fa92bea4244b07a109378f8af3749db8a842d3d: [(RwdMember,DbLovelace 539281)] /= []
cardano-db-sync-extended_1  | [db-sync-node:Error:277] [2022-01-05 23:59:05.00 UTC]   e10168e8b1ae9134a3926ef3d7e8dd3a750542a0579e7cc8dca73b54f4: [(RwdMember,DbLovelace 27670)] /= []
cardano-db-sync-extended_1  | [db-sync-node:Error:277] [2022-01-05 23:59:05.00 UTC]   e10175b7ce7c5a0b94836c9b253b65d69c36b3817572f4642dcdacb241: [(RwdMember,DbLovelace 30944279)] /= []
cardano-db-sync-extended_1  | [db-sync-node:Error:277] [2022-01-05 23:59:05.00 UTC]   e101a789c32cc98fd0321398dfb65b48975358d5a2003505c2a81da166: [(RwdMember,DbLovelace 54299)] /= []
cardano-db-sync-extended_1  | [db-sync-node:Error:277] [2022-01-05 23:59:05.00 UTC]   e101ae99e1492596e5c09ccfa68594612eca5149c8f69c64b8c8ff3791: [(RwdMember,DbLovelace 96251)] /= []
cardano-db-sync-extended_1  | [db-sync-node:Error:277] [2022-01-05 23:59:05.00 UTC]   e101f527120fc00134443d8f4c5c4bba7f5e09876538172cfec95481bb: [(RwdMember,DbLovelace 990700)] /= []
cardano-db-sync-extended_1  | [db-sync-node:Error:277] [2022-01-05 23:59:05.00 UTC]   e10206a118de19d0e9e3b071ae50e213676d133384547031b74ef02bda: [(RwdMember,DbLovelace 1789028)] /= []
cardano-db-sync-extended_1  | [db-sync-node:Error:277] [2022-01-05 23:59:05.00 UTC]   e1021b148e8582e50cc2e988c4af89a27676910ad39ef73de26aee3d3c: [(RwdMember,DbLovelace 2413)] /= []
cardano-db-sync-extended_1  | [db-sync-node:Error:277] [2022-01-05 23:59:05.00 UTC]   e1023549089aac6d0cf69d52276084a3a8e9dd4fe2f5b70a39bb9ed0f0: [(RwdMember,DbLovelace 20506)] /= []
```

**Comment by erikd** [comment]
Check the `db-sync` versions being used. `11.0.*` had a problem. There should be no problem with `12.0.*` . I have an instance running `12.0.1` and it is currently at block `6720761` and running as expected.

**Comment by CyberCyclone** [comment]
All the nodes are running the latest `cardano-graphql` and are running `db-sync` `12.0.0`. So far I haven't seen any projects using just `cardano-db-sync` having any issues, so I think this is specifically cardano-graphql related.

I've just done a query:
```
select block_no from block where block_no is not null
    order by block_no desc limit 1 ;
```

Result: 6720356

Which is the last block of epoch 312. So there does appear to be an issue crossing the boundary but since there's no errors there's not much to go on.

I've tried restarting the Docker stack on one node which doesn't seem to fix it, so on another node I'm trying a restore from the db-sync snapshot and seeing if it can cross the boundary with it.

I'll leave the 3rd node as is incase there's something I can think of to look into.

**Comment by CyberCyclone** [comment]
It seems that restarting the stack does allow it to fix itself and is now in-sync with epoch 313. It's an inconvenient solution but at least there is a way to fix it until we know the cause.

**Comment by erikd** [comment]
Is it possible you are running out of memory? `db-sync` is incredibly memory hungry and you are running that, `cardano-node` and `graphql` all on the same machine. I would however expect all machines running the same stack to have basically the same memory requirements. Are these machines all similarly speced?

**Comment by CyberCyclone** [comment]
They're all running on 64GB of RAM servers but they're inside LXD containers restricted to 32GB of RAM. They're all running with the exact same setup except the server that stayed in sync. That server is using Docker directly with no LXD container setup. They've been running in this configuration for about a month.

I only looked at one servers resources and it didn't look like it was having RAM issues. 

I've renamed the title as it technically wasn't a crash, they simply stopped processing the blocks without any reason.

I'll keep a closer eye on them at the next epoch to see if it happens again and how much resources it uses.
